share/harvest/base.py

```diff
@@ -147,15 +146,17 @@ def fetch_date_range(self, start, end, limit=None, **kwargs):
             )
             start, end = self.shift_range(start, end)
 
-        data_gen = self._do_fetch(start, end, **kwargs)
+        data_gen = self._do_fetch(start, end)
 
         if not isinstance(data_gen, types.GeneratorType) and len(data_gen) != 0:
             raise TypeError('{!r}._do_fetch must return a GeneratorType for optimal performance and memory usage'.format(self))
 
         for i, blob in enumerate(data_gen):
             result = FetchResult(blob[0], self.serializer.serialize(blob[1]), *blob[2:])
 
-            if result.datestamp and (result.datestamp.date() < start.date() or result.datestamp.date() > end.date()):
+            if result.datestamp is None:
+                result.datestamp = end
```

Comment:
I wanted a non-null datestamp so we could reliably figure out the most recent RawDatum for a suid. If the transformer didn't give a useful datestamp, it seemed like a safe assumption the datum was relevant within the harvested range, rather than using `now` or something. Would `start` be better?